queries.sql

```diff
@@ -123,4 +123,76 @@ FROM animals
 JOIN owners ON animals.owner_id = owners.id 
 GROUP BY owners.full_name 
 ORDER BY COUNT(*) DESC 
+
+-- Who was the last animal seen by William Tatcher?
+SELECT a.name 
+FROM animals a 
+JOIN visits v ON a.id = v.animal_id 
+JOIN vets vt ON v.vet_id = vt.id 
+WHERE vt.name = 'William Tatcher' 
+ORDER BY v.visit_date DESC 
+LIMIT 1;
+
+-- How many different animals did Stephanie Mendez see?
+SELECT COUNT(DISTINCT v.animal_id) 
+FROM visits v 
+JOIN vets vt ON v.vet_id = vt.id 
+WHERE vt.name = 'Stephanie Mendez';
+
+-- List all vets and their specialties, including vets with no specialties.
+SELECT vt.name, s.name AS specialty
+FROM vets vt
+LEFT JOIN specializations sp ON vt.id = sp.vet_id
+LEFT JOIN species s ON s.id = sp.species_id
+ORDER BY vt.name;
+
+-- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+SELECT a.name 
+FROM animals a 
+JOIN visits v ON a.id = v.animal_id 
+JOIN vets vt ON v.vet_id = vt.id 
+WHERE vt.name = 'Stephanie Mendez' 
+AND v.visit_date BETWEEN '2020-04-01' AND '2020-08-30';
```

Comment:
- [ ] This query is also returning incorrect result due to incorrect data in the visits table. Kindly fix data and add a new screenshot of correct result.